torch/_inductor/lowering.py

```diff
@@ -185,22 +189,36 @@ def tag_to_layout_constraint(tag):
     raise AssertionError(f"Unknown layout constraint tag: {tag}")
 
 
-def assert_nyi(cond, msg):
+def assert_nyi(cond: bool, msg: str) -> None:
     if not cond:
         raise NotImplementedError(f"inductor does not support {msg}")
 
 
-def add_needs_realized_inputs(fn):
+def add_needs_realized_inputs(
+    fn: Union[
+        list[Union[torch._ops.OpOverload, torch._ops.OpOverloadPacket]],
+        OrderedSet[Union[torch._ops.OpOverload, torch._ops.OpOverloadPacket]],
+        tuple[Union[torch._ops.OpOverload, torch._ops.OpOverloadPacket]],
+        torch._ops.OpOverload,
+        torch._ops.OpOverloadPacket,
+    ],
+) -> Optional[list[Any]]:
     if isinstance(fn, (list, set, tuple, OrderedSet)):  # noqa: set_linter
         return [add_needs_realized_inputs(x) for x in fn]
-    needs_realized_inputs.add(fn)
+    if isinstance(fn, torch._ops.OpOverload):
```

Comment:
Oh, this changed behavior! Was there a bug before?

Reply:
All your comments are good. This one is particularly hmm. You're probably right with your worries!

I was about to say that I did this, but in fact, I didn't. Someone else created this `TypeAlias` at the same time I had another pull request with a very similar `TypeAlias`, sadly, not any better named.

You could consider renaming it to `OpOverloadTypes` in this pull request...?